File: 10518_0516316.cpp
#include<iostream>
#include<vector>
using namespace std;
int main()
{
    int times;
    cin>>times;
    unsigned long long nth;
    int len;
    unsigned long long fac[21];
    fac[0]=1;
    fac[1]=1;
    for(int i=2; i<=20; i++)
    {
            fac[i]=fac[i-1]*i;
    }
    while(times--)
    {
        cin>>len>>nth;
        nth-=1;
        int series[len+1]={0};
        for(int i=1; i<=len; i++)
            series[i]=i;

        int ans[len]={0};
        int now=0;
        if(len>21)
        {
            for(int i=len; i>21; i--)
            {
                ans[now++]=series[1];
                for(int k=1;k<len;k++)
                series[k]=series[k+1];
            }
        }
        int k=len;
        if(len>21)
            k=21;
        for(int i=k; i>=1; i--)
        {
            long long n=nth/fac[i-1];
            ans[now++]=series[1+n];
            for(int k=1+n;k<len;k++)
                series[k]=series[k+1];
            nth%=fac[i-1];
        }
        cout<<'(';
        for(int i=0;i<len;i++)
        {
            if(i!=len-1)
            cout<<ans[i]<<',';
        }
        cout<<ans[len-1]<<')'<<endl;


    }
}

File: 10518_0516316.py
fac=[]
fac.append(1)
fac.append(1)
for _ in range (2,64):
    fac.append(fac[_-1]*_)
times=int(input())
for _ in range(1,times+1):
     [len,nth]=map(int,input().split())
     
     series=list(range(1,len+1))
     ans=[]
     nth-=1
     for k in range (1,len):
         n=(nth)//fac[len-k]
         a=series[n]
         ans.append(a)
         series.remove(a)
         nth%=fac[len-k]
     ans.append(series[0])
     print('(',end='')
     for i in range (0,len-1):
         print("%d,"%ans[i],end='')
     print('%d)'%ans[len-1],end='')
     print('')
     


